### PR TITLE
don't fail on empty filecaches, give empty timestamp

### PR DESCRIPTION
**Pull request by JensTimmerman**

### CODE REVIEW

File: lib/vsc/utils/timestamp.py [hunk 1]
```diff
@@ -105,7 +106,11 @@ def read_timestamp(filename):
 
     """
     cache = FileCache(filename)
-    (_, timestamp) = cache.load('timestamp')
+    try:
+        (_, timestamp) = cache.load('timestamp')
+    except TypeError:
+        logging.warning('could not load timestamp from cache file %s', filename)
+        timestamp = None
```
Comment (itkovian) [inline lib/vsc/utils/timestamp.py]: And the effect for scripts using this?

Comment (stdweird) [reply lib/vsc/utils/timestamp.py]: also, why would a filecache ever be empty? this looks like a serious error, not some harmless warning

Comment (stdweird) [reply lib/vsc/utils/timestamp.py]: then add test for existence, to distinguish between corrupt and never initialised

**Comment by JensTimmerman** [comment]
nothing is using this yet, because https://github.com/hpcugent/vsc-ldap/pull/36 isn't merged yet.

yes, the load returns None, giving a TypeError because the load_timestamp tries to unpack the None, maybe I should just fix the load to return `(None, None)`

File: lib/vsc/utils/timestamp.py (hunk 1 above)
Comment (itkovian) [reply lib/vsc/utils/timestamp.py]: If there's nothing in it, it will be empty as well. Not sure how to check for corrupt.

**Comment by epauwels** [comment]
Ping @itkovian

**Comment by JensTimmerman** [comment]
It doesn't exist on first run, then cache returns None, and this unpack fails.
This should be handled differently, either give an error somewhere else or initialize cleanly on first run.
So the error should happen either in cache, or in the application using this, but not in this place imho. It's very confusing to see this fail on an unpack because you got a None from the cache

**Comment by wdpypere** [comment]
@itkovian what will we doe with this?

**Comment by itkovian** [comment]
I think this can be merged.

**Pull request merged**